packages/autorest.python/autorest/codegen/models/client.py

```diff
@@ -225,6 +225,17 @@ def _imports_shared(self, async_mode: bool) -> FileImport:
                 f"{async_prefix}ARMAutoResourceProviderRegistrationPolicy",
                 ImportType.SDKCORE,
             )
+
+        # import for "typing.Self"
+        file_import.add_import("sys", ImportType.STDLIB)
+        file_import.add_submodule_import(
+            "typing_extensions",
+            "Self",
+            ImportType.BYVERSION,
```

Comment:
Also, conditional import from typing-extensions is an anti-pattern, if a type is not available in all Python versions, we should always import from typing-extensions. Typing-extensions is taking care of the conditional import for us:

https://github.com/python/typing_extensions/blob/53bcdded534494674f893112f71d3be344d65363/src/typing_extensions.py#L2113-L2117

Reply:
@msyyc there might be other places we do conditional imports from typing_extensions, can you make sure to fix those as well? thanks!

Reply:
I see. Let us update the conditional imports of typing_extensions for all places.

Reply:
> @msyyc there might be other places we do conditional imports from typing_extensions, can you make sure to fix those as well? thanks!

Sure, let me check it.

Reply:
Updated